src/sdk/sql_cluster_router.cc

```diff
@@ -261,7 +261,86 @@ std::shared_ptr<SQLInsertRow> SQLClusterRouter::GetInsertRow(const std::string&
+        return false;
+    }
+    ::hybridse::node::NodeManager nm;
+    ::hybridse::plan::PlanNodeList plans;
+    bool ok = GetSQLPlan(sql, &nm, &plans);
+    if (!ok || plans.size() == 0) {
+        LOG(WARNING) << "fail to get sql plan with sql " << sql;
+        status->msg = "fail to get sql plan with";
+        return false;
+    }
+    ::hybridse::node::PlanNode* plan = plans[0];
+    if (plan->GetType() != hybridse::node::kPlanTypeInsert) {
+        status->msg = "invalid sql node expect insert";
+        LOG(WARNING) << "invalid sql node expect insert";
+        return false;
+    }
+    ::hybridse::node::InsertPlanNode* iplan = dynamic_cast<::hybridse::node::InsertPlanNode*>(plan);
+    const ::hybridse::node::InsertStmt* insert_stmt = iplan->GetInsertNode();
+    if (insert_stmt == NULL) {
+        LOG(WARNING) << "insert stmt is null";
+        status->msg = "insert stmt is null";
+        return false;
+    }
+    *table_info = cluster_sdk_->GetTableInfo(db, insert_stmt->table_name_);
+    if (!(*table_info)) {
+        status->msg = "table with name " + insert_stmt->table_name_ + " in db " + db + " does not exist";
+        LOG(WARNING) << status->msg;
+        return false;
+    }
+    std::map<uint32_t, uint32_t> column_map;
+    for (size_t j = 0; j < insert_stmt->columns_.size(); ++j) {
+        const std::string& col_name = insert_stmt->columns_[j];
+        bool find_flag = false;
+        for (int i = 0; i < (*table_info)->column_desc_size(); ++i) {
+            if (col_name == (*table_info)->column_desc(i).name()) {
+                if (column_map.count(i) > 0) {
+                    status->msg = "duplicate column of " + col_name;
+                    LOG(WARNING) << status->msg;
+                    return false;
+                }
+                column_map.insert(std::make_pair(i, j));
+                find_flag = true;
+                break;
+            }
+        }
+        if (!find_flag) {
+            status->msg = "can't find column " + col_name + " in table " + (*table_info)->name();
+            LOG(WARNING) << status->msg;
+            return false;
+        }
+    }
+    for (int i = 0; i < insert_stmt->values_.size(); i++) {
+        hybridse::node::ExprNode* value = insert_stmt->values_[i];
+        if (value->GetExprType() != ::hybridse::node::kExprList) {
+            status->msg =
+                "invalid row expression, expect kExprList but " + hybridse::node::ExprTypeName(value->GetExprType());
+            LOG(WARNING) << status->msg;
+            return false;
+        }
+        uint32_t str_length = 0;
+        default_maps->push_back(GetDefaultMap(*table_info, column_map,
+                                              dynamic_cast<::hybridse::node::ExprListNode*>(value),
+                                              &str_length));
+        if (!default_maps->back()) {
+            status->msg = "get default value map of row[" + std::to_string(i) + "] failed";
```

Comment:
I think It would be better to contain the index info of row in the error message - e.g. "get default value map of row [xxx] ({index}/{total}) failed", but not sure the current format is good enough.

Reply:
Good idea! I will fix it. Thanks.